src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -455,19 +455,18 @@ private TypeSymbol BindTupleType(TupleTypeSyntax syntax, DiagnosticBag diagnosti
 
             if (hasExplicitNames)
             {
-                // If the tuple type with names is bound in a declaration
-                // context then we must have the TupleElementNamesAttribute to emit
-                if (syntax.IsTypeInContextWhichNeedsTupleNamesAttribute())
-                {
-                    // Report diagnostics if System.String doesn't exist
-                    this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+                // If the tuple type with names is bound we must have the TupleElementNamesAttribute to emit
+                // it is typically there though, if we have ValueTuple at all
+                // and we need System.String as well
 
-                    if (!Compilation.HasTupleNamesAttributes)
-                    {
-                        var info = new CSDiagnosticInfo(ErrorCode.ERR_TupleElementNamesAttributeMissing,
-                            AttributeDescription.TupleElementNamesAttribute.FullName);
-                        Error(diagnostics, info, syntax);
-                    }
+                // Report diagnostics if System.String doesn't exist
+                this.GetSpecialType(SpecialType.System_String, diagnostics, syntax);
+
+                if (!Compilation.HasTupleNamesAttributes)
```

Comment:
Tuple literals, or tuple typed expressions in general, cannot result in a declaration requiring this attribute (in regular, not scripting code). 
For now we have agreed on requiring this check only on tuple type syntax - simple and prevents inconvenient situations where we have to emit the attribute and do not have one. 

That may not be the final state of the rule, but that is what this change is implementing. 


---
In reply to: [80973604](https://github.com/dotnet/roslyn/pull/14023#discussion_r80973604) [](ancestors = 80973604)